### ISSUE TITLE
Can't build archlinux template because vmm-xen fails to build (Qubes 4.0)

### ISSUE THREAD

**Issue by danegoodwin** [QubesOS/qubes-issues#4362]
### Qubes OS version:

R4.0

### Affected component: 

vmm-xen when building archlinux
(template used for building: fedora-26)

---

### Steps to reproduce the behavior:

Try to build the archlinux template:

1. Install and set up the <a href="https://www.qubes-os.org/doc/qubes-builder/">needed prerequisites</a> for building QubesOS
2. Clone <a href="https://github.com/QubesOS/qubes-builder">qubes-builder</a>
3. Copy <a href="https://github.com/QubesOS/qubes-builder/blob/master/example-configs/qubes-os-master.conf">example-configs/qubes-os-master.conf</a> as builder.conf
4. Set `DISTS_VM` to `archlinux` and add `builder-archlinux` to `COMPONENTS` and `BUILDER_PLUGINS`
5. Run `make get-sources`, `make qubes-vm`. The error I get happens here, before I run `make template`.

### Expected behavior:

The packages should build correctly so I can create an archlinux template.

### Actual behavior:

`vmm-xen` fails to build with the following error:
```
/home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools/libxl/mk_dsdt --debug=n --maxcpu any --dm-version none >> /home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools/libxl/dsdt_pvh.asl
iasl -vs -p /home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools/libxl/dsdt_pvh.tmp -tc /home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools/libxl/dsdt_pvh.asl
free(): invalid pointer
make[4]: *** [Makefile:78: /home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools/libxl/dsdt_pvh.c] Aborted (core dumped)
make[4]: Leaving directory '/home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools/libacpi'
make[3]: *** [Makefile:89: acpi] Error 2
make[3]: Leaving directory '/home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools/libxl'
make[2]: *** [/home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools/../tools/Rules.mk:218: subdir-install-libxl] Error 2
make[2]: Leaving directory '/home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools'
make[1]: *** [/home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools/../tools/Rules.mk:213: subdirs-install] Error 2
make[1]: Leaving directory '/home/user/qubes-src/vmm-xen/src/xen-4.8.4/tools'
make: *** [Makefile:101: install-tools] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
make[2]: *** [/home/user/qubes-builder/qubes-src/builder-archlinux/Makefile.archlinux:122: dist-package] Error 4
```

### General notes:

Here are my [builder.conf](https://github.com/QubesOS/qubes-issues/files/2438771/builder.conf.txt) and [vmm-xen-vm-archlinux.log](https://github.com/QubesOS/qubes-issues/files/2438772/vmm-xen-vm-archlinux.log.txt).

I think I set everything up correctly. During the build process, pacman runs and downloads all the relevant packages, but `vmm-xen` doesn't build and the process stops there. 

---

### Related issues:

None that I could find.

**Comment by NegativeFeedback** [comment]
I think I ran into the same issue, tried some suggestions on the reddit thread I asked for help in. I will post when back at my desk. Ended up with a fresh install as I fubared up the install.

**Comment by CharlesDardaman** [comment]
@NegativeFeedback Hey did you ever get back to your desk?

**Comment by marmarek** [comment]
@ptitdoc what Xen package version is available in native Arch? Anything >= 4.8 (but the newer the better) should be enough to use native packaging, instead of qubes custom one - all the changes really needed in qubes VM are already upstream.
As for the packaging, common problems:
 - libxenvchan library included (was missing in both native Fedora and Debian packages for a long time)
 - xendriverdomain.service packaged (still missing in Debian packaging)
 - do _not_ try to initialize dom0 services (xenstored, xenconsoled etc) services when running in guest (broken in Debian, there is a chance to fix in buster)

In Qubes 4.1, we [will use native Fedora packages](https://github.com/QubesOS/qubes-vmm-xen/pull/51) for guests. There is also a chance that relevant changes will land in buster in time, so we'll drop custom packaging there too. If Arch template could join here, that would be great.

@CharlesDardaman @NegativeFeedback @danegoodwin you can try by removing `vmm-xen` from `COMPONENTS` (or `TEMPLATE` depending whether you use `setup` script or not) setting in `builder.conf` and try building the template. Please report any build and runtime errors you got.

**Comment by danegoodwin** [comment]
@andrewdavidwong We can mark this as resolved. I rebuilt my builder appvm to try @marmarek's recommendation, but arch built just fine, so my best guess is that the dependency issue got resolved upstream between then and now. I'm now the happy owner of a fully functioning arch template!
@NegativeFeedback I hope you're OK dude

**Comment by marmarek** [comment]
@danegoodwin did your build succeeded including vmm-xen? Or have you excluded it?
Trying native Arch Xen packages would still be useful, as that would be one less thing to maintain. For example Arch packaging in xen-4.12 branch (for upcoming R4.1) is currently broken...

**Comment by ptitdoc** [comment]
@marmarek : I will test out the packaging and come back to you if it works in Qubes 4.0.

Sadly, xen is not available anymore as a binary package in Archlinux, meaning we have to build it anyway. Currently available PKGBUILD are using Xen 4.11 as it is the official last xen version, and is building the whole hypervisor and tools.

We could still do a PKGBUILD for a native xen build (VM tools only) and if there are enough votes, it would be promoted to binary package.
